Framework/Algorithms/src/CalculateIqt.cpp

```diff
@@ -0,0 +1,309 @@
+  positiveInt->setLower(1);
+
+  declareProperty("NumberOfIterations", DEFAULT_ITERATIONS, positiveInt,
+                  "Number of randomised simulations within error to run.");
+  declareProperty(
+      "SeedValue", DEFAULT_SEED, positiveInt,
+      "Seed the random number generator for monte-carlo error calculation.");
+
+  declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                   Direction::Output),
+                  "The name to use for the output workspace.");
+}
+
+void CalculateIqt::exec() {
+  const auto rebinParams = rebinParamsAsString();
+  const MatrixWorkspace_sptr sampleWorkspace = getProperty("InputWorkspace");
+  MatrixWorkspace_sptr resolution = getProperty("ResolutionWorkspace");
+  resolution = normalizedFourierTransform(resolution, rebinParams);
+
+  // create function which only needs sampleWorkspace as input so it can be
+  // easily repeated in simulations
+  auto calculateIqtFunction =
+      [resolution, &rebinParams](MatrixWorkspace_sptr sample) {
+        return calculateIqt(sample, resolution, rebinParams);
+      };
+
+  auto outputWorkspace = monteCarloErrorCalculation(sampleWorkspace, resolution,
+                                                    calculateIqtFunction);
+
+  outputWorkspace = cleanOutput(outputWorkspace);
+  setProperty("OutputWorkspace", outputWorkspace);
+}
+
+std::string CalculateIqt::rebinParamsAsString() {
+  const double e_min = getProperty("EnergyMin");
+  const double e_max = getProperty("EnergyMax");
+  const double e_width = getProperty("EnergyWidth");
+  return createRebinString(e_min, e_max, e_width);
+}
+
+MatrixWorkspace_sptr CalculateIqt::monteCarloErrorCalculation(
+    MatrixWorkspace_sptr sample, MatrixWorkspace_sptr resolution,
+    const std::function<MatrixWorkspace_sptr(MatrixWorkspace_sptr)> &
+        calculateIqtFunction) {
+  auto outputWorkspace = calculateIqtFunction(sample);
+  const unsigned int nIterations = getProperty("NumberOfIterations");
+  const unsigned int seed = getProperty("SeedValue");
+  std::vector<MatrixWorkspace_sptr> simulatedWorkspaces;
+  simulatedWorkspaces.reserve(nIterations);
+  simulatedWorkspaces.emplace_back(outputWorkspace);
+
+  PARALLEL_FOR_IF(Kernel::threadSafe(*sample, *resolution))
+  for (auto i = 0; i < nIterations - 1; ++i) {
+    PARALLEL_START_INTERUPT_REGION
+    auto simulatedWorkspace =
+        randomizeWorkspaceWithinError(sample->clone(), seed);
+    simulatedWorkspace = calculateIqtFunction(simulatedWorkspace);
+    simulatedWorkspaces.emplace_back(simulatedWorkspace);
+    PARALLEL_END_INTERUPT_REGION
+  }
+  PARALLEL_CHECK_INTERUPT_REGION
+  return setErrorsToStandardDeviation(nIterations, simulatedWorkspaces,
+                                      outputWorkspace);
+}
+
+MatrixWorkspace_sptr CalculateIqt::setErrorsToStandardDeviation(
+    int nIterations,
+    const std::vector<MatrixWorkspace_sptr> simulatedWorkspaces,
```

Comment:
Why is this a method rather than free function in the cpp file?
It doesn't appear to require any class fields or methods.

Also missing a & here.
```const std::vector<MatrixWorkspace_sptr> &simulatedWorkspaces```